onnxruntime/core/platform/path_lib.h

```diff
@@ -178,11 +178,12 @@ inline OrtFileType DTToFileType(DWORD dwFileAttributes) {
   return OrtFileType::TYPE_REG;
 }
 inline std::string FormatErrorCode(DWORD dw) {
-  char* lpMsgBuf;
-  FormatMessageA(FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS, NULL, dw,
-                 MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), (LPSTR)&lpMsgBuf, 0, NULL);
+  static const DWORD bufferLength = 64 * 1024;
+  char * lpMsgBuf = new char[bufferLength];
+  FormatMessageA(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS, NULL, dw,
+                 MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), (LPSTR)lpMsgBuf, bufferLength / sizeof(TCHAR), NULL);
   std::string s(lpMsgBuf);
```

Comment:
Nit: you may use the buffer from s to reduce one heap allocation.
```
std::string s(64);
FormatMessageA(...,(LPSTR)s.data(),...)
```

Reply:
Went ahead and changed this, can you reapprove? thanks!